service/orchestrator/cloud_service.go

```diff
@@ -87,9 +89,18 @@ func (svc *Service) ListCloudServices(_ context.Context, req *orchestrator.ListC
 		return
 	}
 
+	res = new(orchestrator.ListCloudServicesResponse)
+
+	// Retrieve list of allowed cloud service according to our authorization strategy. No need to specify any conditions
+	// to our storage request, if we are allowed to see all cloud services.
+	all, allowed = svc.authz.AllowedCloudServices(ctx)
+	if !all {
+		conds = append(conds, allowed)
```

Comment:
Why the appending works here without the variadic thing: `slice = append(slice, anotherSlice...)`?
They are of different type anyway

Reply:
because I want to end up with the array as an argument and not with the individual elements of the array. In the end we should have one element in cond (an array) and thus we trigger the gorm function that uses this element to query the primary key.